src/ts/Html5Metadata.ts

```diff
@@ -1,6 +1,11 @@
 import Client = Conviva.Client;
 
 export class Html5Metadata implements Conviva.MetadataInterface {
+  deviceCategory?: Client.DeviceCategory;
+
+  constructor(deviceCategory?: Client.DeviceCategory) {
```

Comment:
Is there a particular reason to have the DeviceCategory as a constructor argument instead of just as a settable property in this class?

Reply:
Not necessarily, only to make instantiating the class a bit easier. Can you think of a reason to not have it in the constructor?

Reply:
Or we could pass it as a config object via the constructor.

Creating an interface `DeviceMetadata` or maybe `ClientMetadata` which has a property `deviceCategory` which is mandatory and pass it via the constructor as a `metadata` property.

The problem I see with the getter is that we are implementing an interface `Conviva.MetadataInterface` which does not define those getters. So we can't implement against that interface when using an instance of `Html5Metadata `